File: CoinKeeper.BusinessLogic/Users/Models/Registration/RegistrationResult.cs
using System.Text.Json.Serialization;

namespace Domain.Entities.Users.Registration;

public class RegistrationResult
{
    [JsonPropertyName("Status")]
    public RegistrationStatus Status { get; set; }
    
    [JsonPropertyName("Message")]
    public string Message { get; set; } = null!;

    [JsonPropertyName("Errors")]
    public IEnumerable<string> Errors { get; set; } = null!;

    [JsonPropertyName("Data")]
    public UserViewModel Data { get; set; } = null!;
}

File: CoinKeeper.Api/ClientApp/src/models/User/RegistrationResult.ts
import {RegistrationStatus} from "./RegistrationStatus";
import {UserViewModel} from "./UserViewModel";

export class RegistrationResult {
    public Status: RegistrationStatus = RegistrationStatus.Error;
    public Message: string = "";
    public Errors: string[] = [];
    public Data: UserViewModel | null = null;
}